src/shared/domain/definition/case-field.model.ts

```diff
@@ -26,20 +26,39 @@ export class CaseField implements Orderable {
   @Type(() => WizardPageField)
   wizardProps?: WizardPageField;
 
+  private _value: any;
+  private _list_items: any = [];
 
-  getValue?(): string {
-    if (this.field_type.type === 'DynamicList') {
-      return this.value ? this.value.value.code : '';
+  @Expose()
+  get value(): any {
+    if (this.field_type && this.field_type.type === 'DynamicList') {
+      return this._value && this._value.value ? this._value.value.code : this._value;
     } else {
-      return this.value;
+      return this._value;
     }
   }
 
-  getItems?(): any {
-    if (this.field_type.type === 'DynamicList') {
-      return this.value ? this.value.list_items : [];
+  set value(value: any) {
+    this._value = value;
+  }
+
+  @Expose()
+  get list_items(): any {
+    if (this.field_type && this.field_type.type === 'DynamicList') {
+      return this._value && this._value.list_items ? this._value.list_items : this._list_items;
     } else {
       return this.field_type.fixed_list_items;
     }
   }
+
+  set list_items(items: any) {
+    this._list_items = items;
+  }
+
+  @Expose()
+  public isReadonly() {
+    return  (!(typeof this.display_context === 'undefined' || this.display_context === null )
```

Comment:
can't this `!(typeof this.display_context === 'undefined' || this.display_context === null )` be replaced simply by:
`this.display_context`?
so 
`display_context && this.display_context.toUpperCase() === 'READONLY')`

Reply:
Unfortunately not,
display_context && this.display_context.toUpperCase() === 'READONLY' would return an undefined.
Its weird how Javascript works., hence the change